tools/hive-bigquery/bigquery_component.py

```diff
@@ -0,0 +1,725 @@
+        table = bigquery.Table(table_ref, schema)
+        self.client.create_table(table)
+
+    def delete_table(self, dataset_id, table_name):
+        """Deletes BigQuery table
+
+        Args:
+            dataset_id (str): BigQuery dataset id
+            table_name (str): BigQuery table name
+        """
+
+        table_ref = self.client.dataset(dataset_id).table(table_name)
+        try:
+            self.client.delete_table(table_ref)
+            logger.debug(
+                "Deleted table %s from %s dataset", table_name, dataset_id)
+        except Exception as error:
+            logger.error(error)
+
+    def get_table(self, dataset_id, table_name):
+        """Gets BigQuery table
+
+        Args:
+            dataset_id (str): BigQuery dataset id
+            table_name (str): BigQuery table name
+
+        Returns:
+            google.cloud.bigquery.table.Table: BigQuery table instance
+        """
+
+        table_ref = self.client.dataset(dataset_id).table(table_name)
+        table = self.client.get_table(table_ref)
+        return table
+
+    def check_bq_write_mode(self, mysql_component, hive_table_model,
+                            bq_table_model):
+        """Validates the bq_table_write_mode provided by user
+
+        If the mode is overwrite, drops the tracking table and deletes the
+        BigQuery table. If the mode is create, checks if the tracking table
+        and BigQuery table exist. If the mode is append, checks whether the
+        BigQuery table exists
+
+        Args:
+            mysql_component (:class:`MySQLComponent`): Instance of
+                MySQLComponent to connect to MySQL
+            hive_table_model (:class:`HiveTableModel`): Wrapper to Hive table
+                details
+            bq_table_model (:class:`BigQueryTableModel`): Wrapper to BigQuery
+                table details
+        """
+
+        if PropertiesReader.get('bq_table_write_mode') == "overwrite":
+            logger.debug("Deleting tracking table and BigQuery table...")
+            mysql_component.drop_table(hive_table_model.tracking_table_name)
+            self.delete_table(bq_table_model.dataset_id,
+                              bq_table_model.table_name)
+            hive_table_model.is_first_run = True
+
+        elif PropertiesReader.get('bq_table_write_mode') == "create":
+            if not hive_table_model.is_first_run:
+                print_and_log("Tracking Table {} already exist".format(
+                    hive_table_model.tracking_table_name), logging.CRITICAL)
+                exit()
+            if self.check_bq_table_exists(bq_table_model.dataset_id,
+                                          bq_table_model.table_name):
+                print_and_log(
+                    "BigQuery Table {} already exist in {} dataset".format(
+                        bq_table_model.table_name, bq_table_model.dataset_id),
+                    logging.CRITICAL)
+
+                exit()
```

Comment:
I see that we are using exit() to break the program at many places. Is there a way to avoid this at all? Perhaps encapsulate further into smaller functions with appropriate return value.